src/coreclr/tools/aot/ILCompiler.Compiler/IL/ILImporter.Scanner.cs

```diff
@@ -172,9 +182,21 @@ public DependencyList Import()
             FindBasicBlocks();
             ImportBasicBlocks();
 
-            CodeBasedDependencyAlgorithm.AddDependenciesDueToMethodCodePresence(ref _dependencies, _factory, _canonMethod, _canonMethodIL);
+            CombinedDependencyList conditionalDependencies = null;
+            foreach (BasicBlock bb in _basicBlocks)
```

Comment:
Why can there be `null` blocks in this list?

Reply:
This array is indexed into by the IL offset. The entries are non-null at basic block start locations and null elsewhere.